reframe/core/hooks.py

```diff
@@ -170,10 +170,15 @@ def __contains__(self, key):
     def __getattr__(self, name):
         return getattr(self.__hooks, name)
 
-    def update(self, hooks):
+    def update(self, hooks, *, denied_hooks={}):
```

Comment:
```suggestion
    def update(self, hooks, *, denied_hooks=None):
```

And set it in the body with `denied_hooks = denied_hooks or set()`. Otherwise (a) the default arg. is not a set and (b) more importantly, the default argument is mutable here.

Reply:
Oh my ... what a rookie mistake 🙈!